third_party/lua/luadebug.lua

```diff
@@ -248,14 +248,34 @@ end
 local SOURCE_CACHE = {}
 
 local function where(info, context_lines)
+    local filesource = info.source
     local source = SOURCE_CACHE[info.source]
     if not source then
         source = {}
-        local filename = info.source:match("@(.*)")
-        if filename then
-            pcall(function() for line in io.lines(filename) do table.insert(source, line) end end)
-        elseif info.source then
-            for line in info.source:gmatch("(.-)\n") do table.insert(source, line) end
+        -- Tarantool builtin module
+        if filesource:match("@builtin/.*.lua") then
+            pcall(function()
+                local tnt_debug = require('tarantool').debug
```

Comment:
Excess variable: see no benefits in its usage.

Reply:
But after this commit you don't check whether `tnt` is initialized.

Reply:
Good point - added check of `tnt` at the moment of require
```diff
diff --git a/third_party/c-dt b/third_party/c-dt
--- a/third_party/c-dt
+++ b/third_party/c-dt
@@ -1 +1 @@
-Subproject commit cec6acebb54d9e73ea0b99c63898732abd7683a6
+Subproject commit cec6acebb54d9e73ea0b99c63898732abd7683a6-dirty
diff --git a/third_party/lua/luadebug.lua b/third_party/lua/luadebug.lua
index 700e88a80..c0f341077 100644
--- a/third_party/lua/luadebug.lua
+++ b/third_party/lua/luadebug.lua
@@ -620,6 +620,7 @@ local function motto()
     -- Detect Tarantool version.
     if not tnt then
         tnt = require('tarantool')
+        assert(tnt ~= nil)
     end
     LUA_JIT_SETLOCAL_WORKAROUND = -1
     dbg_writeln(color_yellow(DEBUGGER .. ": ") .. "Loaded for " .. tnt.version)
```

Reply:
Anyway, it doesn't solve the issue when `tnt` is accessed before being initialized.

Reply:
`motto` is a delay load for initialization, it's called the first time debugger instantiated. There is no known (at least for me) control flow which could led to `where` invocation, without calling `motto`.

Reply:
I don't know what to say more: if you want to leave this with no change, I don't have much time to argue. Let's proceed with the patch then.